database/neo4j/neo4j.go

```diff
@@ -0,0 +1,202 @@
+		return nil, err
+	}
+
+	return driver, nil
+}
+
+func (n *Neo4j) Open(url string) (database.Driver, error) {
+	uri, err := neturl.Parse(url)
+	if err != nil {
+		return nil, err
+	}
+	password, _ := uri.User.Password()
+	authToken := neo4j.BasicAuth(uri.User.Username(), password, "")
+
+	return WithInstance(&Config{
+		URL:             url,
+		AuthToken:       authToken,
+		MigrationsLabel: DefaultMigrationsLabel,
+	})
+}
+
+func (n *Neo4j) Close() error {
+	return n.driver.Close()
+}
+
+// local locking in order to pass tests, Neo doesn't support database locking
+func (n *Neo4j) Lock() error {
+	if !atomic.CompareAndSwapUint32(&n.lock, 0, 1) {
+		return database.ErrLocked
+	}
+
+	return nil
+}
+
+func (n *Neo4j) Unlock() error {
+	atomic.StoreUint32(&n.lock, 0)
+	return nil
+}
+
+func (n *Neo4j) Run(migration io.Reader) error {
+	body, err := ioutil.ReadAll(migration)
+	if err != nil {
+		return err
+	}
+
+	session, err := n.driver.Session(neo4j.AccessModeWrite)
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	result, err := session.Run(string(body[:]), nil)
+	if err != nil {
+		return err
+	}
+	return result.Err()
+}
+
+func (n *Neo4j) SetVersion(version int, dirty bool) error {
+	session, err := n.driver.Session(neo4j.AccessModeRead)
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	query := fmt.Sprintf("MERGE (sm:%s {version: $version, dirty: $dirty})",
+		n.config.MigrationsLabel)
+	result, err := session.Run(query, map[string]interface{}{"version": version, "dirty": dirty})
+	if err != nil {
+		return err
+	}
+	return result.Err()
+}
+
+type MigrationRecord struct {
+	Version int
+	Dirty bool
+}
+
+func (n *Neo4j) Version() (version int, dirty bool, err error) {
+	session, err := n.driver.Session(neo4j.AccessModeRead)
+	if err != nil {
+		return -1, false, err
+	}
+	defer session.Close()
+
+	query := fmt.Sprintf("MATCH (sm:%s) RETURN sm.version AS version, sm.dirty AS dirty ORDER BY sm.version DESC LIMIT 1",
+		n.config.MigrationsLabel)
+	result, err := session.ReadTransaction(func(transaction neo4j.Transaction) (interface{}, error) {
+		result, err := transaction.Run(query, nil)
+		if err != nil {
+			return nil, err
+		}
+		if result.Next() {
+			record := result.Record()
+			mr := MigrationRecord{}
+			versionResult, ok := record.Get("version")
+			if !ok {
+				mr.Version = -1
+			} else {
+				mr.Version = int(versionResult.(int64))
+			}
+
+			dirtyResult, ok := record.Get("dirty")
+			if ok {
+				mr.Dirty = dirtyResult.(bool)
+			}
+
+			return mr, nil
+		}
+		return nil, result.Err()
+	})
+	if err != nil {
+		return -1, false, err
+	}
+	if result == nil {
+		return -1, false, nil
+	}
+	mr := result.(MigrationRecord)
+	return mr.Version, mr.Dirty, nil
+}
+
+func (n *Neo4j) Drop() error {
+	session, err := n.driver.Session(neo4j.AccessModeWrite);
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	_, err = session.Run("MATCH (n) DETACH DELETE n", map[string]interface{}{})
```

Comment:
Be consistent and use `nil` instead of `map[string]interface{}{}`

Reply:
Done